apps/common/lib/lexical/code_unit.ex

```diff
@@ -87,6 +91,19 @@ defmodule Lexical.CodeUnit do
     do_count_utf16(rest, count + increment)
   end
 
+  defp do_count_utf8(<<>>, count) do
+    count
+  end
+
+  defp do_count_utf8(<<c, rest::binary>>, count) when c < 128 do
+    do_count_utf8(rest, count + 1)
+  end
+
+  defp do_count_utf8(<<c::utf8, rest::binary>>, count) do
+    increment = byte_size(<<c::utf8>>)
+    do_count_utf8(rest, count + increment)
+  end
+
   defp do_utf16_offset(_, 0, offset) do
```

Comment:
What is this for? 

is it some special cases that cannot be converted to UTF-8?

Reply:
this is the case where the offset has been found and we return it.